qt/applications/workbench/workbench/app/mainwindow.py

```diff
@@ -364,9 +365,14 @@ def launch_custom_python_gui(self, filename):
         self.interface_executor.execute(open(filename).read(), filename)
 
     def launch_custom_cpp_gui(self, interface_name):
-        interface = self.interface_manager.createSubWindow(interface_name)
-        interface.setAttribute(Qt.WA_DeleteOnClose, True)
-        interface.show()
+        window = find_window(interface_name, QMainWindow)
```

Comment:
I know you didn't add this method but could you add some method documentation to it? Especially now there is more complicated behaviour we should explain that this method creates a new window if one does not already exist else it simply shows the current window.